src/views/ops/Template/components/Variable.vue

```diff
@@ -70,14 +77,15 @@ export default {
   },
   watch: {
     variables: {
-      handler(value) {
-        if (value.length > 0 || this.initial) {
-          value.map((item) => {
+      handler(newVal, oldVal) {
+        if (oldVal === undefined) return
+        if (newVal.length > 0 || !this.initial) {
+          newVal.map((item) => {
             item.default_value = item.text_default_value || item.select_default_value
           })
-          this.$emit('input', value)
+          this.$emit('input', newVal)
         }
-        if (value) {
+        if (newVal) {
           this.initial = true
         }
       },
```

Comment:
I need the actual code to be checked and evaluated for differences, potential issues, or optimization suggestions. Please share the specific sections or elements you'd like me to analyze.